### ISSUE TITLE
Depends on: gtk3 >= 3.14

### ISSUE THREAD

**Issue by bernd-wechner** [gnumdk/lollypop#653]
On Ubuntu and Unbuntu derivatives the latest version of gtk available in package repos is 3.10.8.

Hence impossible to build Lollypop. Would be very useful to get some tips from an experience dev about this, for example:

1) Why depend on 3.14 and not 3.10?
2) What distro are you using and does it come with 3.14 in its repos?
3) Is it sensible to install 3.14 on Ubuntu and if so how?

**Comment by gnumdk** [comment]
On Ubuntu, last gtk available is 3.18

Upgrade your distro.

**Comment by bernd-wechner** [comment]
Thanks for the prompt response. Great to hear.

Have done. Upgrade and update. No luck.

 Can you point to a repo that stocks 3.18? Of the package libgtk-3-dev?

Here is the list of versions form my repos (list version found and repo its in):

```
$ apt-cache showpkg libgtk-3-dev
Package: libgtk-3-dev
Versions: 
3.10.8-0ubuntu1.6 (/var/lib/apt/lists/archive.ubuntu.com_ubuntu_dists_trusty-updates_main_binary-amd64_Packages) (/var/lib/dpkg/status)
 Description Language: 
                 File: /var/lib/apt/lists/packages.linuxmint.com_dists_rosa_upstream_binary-amd64_Packages
                  MD5: 5e247317651e3625eb226ee49f0cfec7
 Description Language: en
                 File: /var/lib/apt/lists/archive.ubuntu.com_ubuntu_dists_trusty_main_i18n_Translation-en
                  MD5: 5e247317651e3625eb226ee49f0cfec7

3.10.8-0ubuntu1.4 (/var/lib/apt/lists/security.ubuntu.com_ubuntu_dists_trusty-security_main_binary-amd64_Packages)
 Description Language: 
                 File: /var/lib/apt/lists/packages.linuxmint.com_dists_rosa_upstream_binary-amd64_Packages
                  MD5: 5e247317651e3625eb226ee49f0cfec7
 Description Language: en
                 File: /var/lib/apt/lists/archive.ubuntu.com_ubuntu_dists_trusty_main_i18n_Translation-en
                  MD5: 5e247317651e3625eb226ee49f0cfec7

3.10.8-0ubuntu1 (/var/lib/apt/lists/archive.ubuntu.com_ubuntu_dists_trusty_main_binary-amd64_Packages)
 Description Language: 
                 File: /var/lib/apt/lists/packages.linuxmint.com_dists_rosa_upstream_binary-amd64_Packages
                  MD5: 5e247317651e3625eb226ee49f0cfec7
 Description Language: en
                 File: /var/lib/apt/lists/archive.ubuntu.com_ubuntu_dists_trusty_main_i18n_Translation-en
                  MD5: 5e247317651e3625eb226ee49f0cfec7

3.10.8~8+qiana (/var/lib/apt/lists/packages.linuxmint.com_dists_rosa_upstream_binary-amd64_Packages)
 Description Language: 
                 File: /var/lib/apt/lists/packages.linuxmint.com_dists_rosa_upstream_binary-amd64_Packages
                  MD5: 5e247317651e3625eb226ee49f0cfec7
 Description Language: en
                 File: /var/lib/apt/lists/archive.ubuntu.com_ubuntu_dists_trusty_main_i18n_Translation-en
                  MD5: 5e247317651e3625eb226ee49f0cfec7
```

**Comment by bil-elmoussaoui** [comment]
@bernd-wechner upgrade and update won't install the latest version of gtk3. You will need to update the whole OS. you can take a look here how to do that http://www.ubuntu.com/download/desktop/upgrade

**Comment by bernd-wechner** [comment]
I'm sorry, but that's is just not easy to believe. On what basis would you claim that, I'm curious. To begin with makes no sense to me.

This I have already done, and routinely do when in doubt:

sudo apt update
sudo apt upgrade

That doesn't even touch the libgtk-3-dev package as far as I can tell, and there has to be way to install a given package like libgtk-3-dev and hey there is:

sudo apt install libgtk-3-dev

and voila:

```
$ sudo apt install libgtk-3-dev
Reading package lists... Done
Building dependency tree       
Reading state information... Done
libgtk-3-dev is already the newest version.
0 to upgrade, 0 to newly install, 0 to remove and 4 not to upgrade.
```
heck you can even upgrade your kernel that way if you're keen and I have installed the latest version of gtk3 that way (got a later version than I had). But in this repo:

http://archive.ubuntu.com/ubuntu

apt is only finding 3.10 as I illustrated above. but browsing it manually I cans see:

http://archive.ubuntu.com/ubuntu/pool/main/g/gtk+3.0/

and there is clearly:

libgtk-3-dev_3.18.9-1ubuntu3_amd64.deb

among many others, so now I only need to know why apt pointing at that repo won't show me more than 3.10.

Looks like I have som research to do, but it would be so much easier if someone who knows helped out. I'm not on Ubuntu anyhow but on Mint but same deal, done a whole update and upgrade and am on latest and Mint is just an Ubuntu redistro.

And I'm on Mint 17.3 fully updated, upgraded latest everything, and "Linux Mint 17.3 is based on Ubuntu 14.04" and the latest Ubuntu release is apparently 16.04 and that may be the root of the problem.

Which simply raises the question: Is it possible to build Lollypop on Ubuntu 14.04 (or Mint 17.3)? And if so is there a real good reason (does it rely on changes in gtk between 3.10 and 3.14?) or is it just a default of some sort not really deeply considered?

**Comment by bil-elmoussaoui** [comment]
Well the latest version of Mint still does not have the latest version of Gtk. Gtk is mainly dev by Gnome. Ubuntu and a lot of other distros still use an older version of Gtk. You can't upgrade Gtk just by doing a normal upgrade (`apt-get update && apt-get upgrade`). In Ubuntu, Mint, Fedora you will need to fully upgrade your OS(linux distro) to a newer version. For mint, the latest version does not have a newer gtk version and still uses 3.10 i guess, Ubuntu 14.04 uses the same version too as mint is based on top of it. You will need either to install Ubuntu 15.04+ or a newer version.
And you can't build Lollypop on Gtk3.10 as Gtk3.10 does not provide widgets used by applications that were only added on Gtk3.14 (the headerbar)

**Comment by bernd-wechner** [comment]
Turns out Mint 18 is based on Ubuntu 16.04 LTS and is due soon:

http://blog.linuxmint.com/?p=3032

So maybe that'll solve my issue. But still, it is a philosophic question raised above, as to the reason for the dependency on a gtk version younger than is available on Mint 17.3 the most popular distro by an arm and a leg:

http://distrowatch.com/dwres.php?resource=popularity

Almost twice as popular as Ubuntu?

**Comment by bil-elmoussaoui** [comment]
The dev can't do anything about it. It's up to the user either to upgrade or change the distro or not using Lollypop.

**Comment by julianrichen** [comment]
Maybe I'm not reading this thread write but it seems you are only upgrading to the latest point release of that gtk version (`upgrade`). You need to `dist-upgrade` to update to newer versions of packages and update dependencies.

```
apt-get update
apt-get upgrade
apt-get dist-upgrade
```

Obviously back-up your data when you upgrade to newer major version because there is always a 0.1% chance something can go wrong.

Or you can follows bil-elmoussaoui link for the graphical version.

Differences (via man & truncated):
````
upgrade
    upgrade is used to install the newest versions of all packages
    currently installed on the system from the sources enumerated in
    /etc/apt/sources.list...
dist-upgrade
    dist-upgrade in addition to performing the function of upgrade,
    also intelligently handles changing dependencies with new versions
    of packages...
```

**Comment by gnumdk** [comment]
I really don't care about mint, ubuntu LTS is now lollypop target.

**Comment by d9h02f** [comment]
Popularity on distrowatch is determined by how many people look it up on Distrowatch, by the way.

**Comment by bernd-wechner** [comment]
gnumdk, well ya lost me to quodlibet. Easy to work on and so far damned impressive with an dev community that doesn't respond with "don't cares" to ney folk keen to keep it alive and growing but rather lend a hand! Each to their own. Just sayin' ...
